app/controllers/steps/dwp_step_controller.rb

```diff
@@ -5,5 +5,11 @@ class DWPStepController < BaseStepController
     def decision_tree_class
       Decisions::DWPDecisionTree
     end
+
+    def benefit_check_on_partner?
+      return true if current_crime_application.partner&.has_passporting_benefit?
```

Comment:
@hiboabd is the DWP check run when the partner has a contrary interest? If not, I think we should add an `include_partner_in_means_assessment?` guard here as well. Even if it is possible to use `has_passporting_benefit` to infer this, I think the code should explicitly use that guard here.